File: app/src/main/java/com/avidprogrammers/insurancepremiumcalculator/WebViewActivity.java
package com.avidprogrammers.insurancepremiumcalculator;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.MenuItem;
import android.webkit.WebChromeClient;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.Toast;

public class WebViewActivity extends AppCompatActivity implements ConnectivityReceiver.ConnectivityReceiverListener{

    WebView webView;
    Bundle bundle;
    String URL;
    ConnectivityReceiver conn;

    CheckingStatus checkingStatus;
    @SuppressLint("JavascriptInterface")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_web_view);
        webView = findViewById(R.id.webview);
        bundle = getIntent().getExtras();

        checkingStatus=new CheckingStatus();
        conn=new ConnectivityReceiver();
        IntentFilter intentFilter=new IntentFilter();
        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION);
        registerReceiver(conn, intentFilter);
        checkfunction(WebViewActivity.this);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        if(bundle != null)
        {
            URL = bundle.getString("url");
            webView.loadUrl(URL);
            webView.getSettings().setJavaScriptEnabled(true);
            webView.getSettings().setLoadWithOverviewMode(true);
            webView.getSettings().setUseWideViewPort(true);
            webView.getSettings().setPluginState(WebSettings.PluginState.ON);
            webView.getSettings().setRenderPriority(WebSettings.RenderPriority.HIGH);
            webView.getSettings().setCacheMode(WebSettings.LOAD_DEFAULT);
            webView.getSettings().setSavePassword(false);
            webView.getSettings().setDomStorageEnabled(true);
            webView.setHorizontalScrollBarEnabled(false);
            webView.addJavascriptInterface(this, "Android");
            webView.setWebChromeClient(new WebChromeClient());
            webView.setWebViewClient(new WebViewClient());
        }
        else
        {
            Toast.makeText(WebViewActivity.this,"Please Try Again",Toast.LENGTH_SHORT).show();
            finish();
        }
    }

    public void checkfunction(Context context){
        boolean isConnected=ConnectivityReceiver.isConnected();
        //notification(isConnected,lp_taxi_upto18pass.this);
        checkingStatus.notification(isConnected,context);


    }

    @Override
    public void onNetworkConnectionChanged(boolean isConnected) {
        checkingStatus.notification(isConnected,this);

    }

    @Override
    protected void onResume() {
        super.onResume();
        MyApplication.getInstance().setConnectivityListener(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(conn);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch(item.getItemId())
        {
            case android.R.id.home:
                finish();
                break;
        }
        return true;
    }
}

File: app/src/main/java/com/avidprogrammers/insurancepremiumcalculator/WebViewActivity.kt
package com.avidprogrammers.insurancepremiumcalculator

import android.annotation.SuppressLint
import android.content.Context
import android.content.IntentFilter
import android.net.ConnectivityManager
import android.os.Bundle
import android.view.MenuItem
import android.webkit.WebChromeClient
import android.webkit.WebSettings
import android.webkit.WebView
import android.webkit.WebViewClient
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.avidprogrammers.insurancepremiumcalculator.ConnectivityReceiver.ConnectivityReceiverListener

class WebViewActivity : AppCompatActivity(), ConnectivityReceiverListener {
    var webView: WebView? = null
    var bundle: Bundle? = null
    var URL: String? = null
    var conn: ConnectivityReceiver? = null
    var checkingStatus: CheckingStatus? = null
    @SuppressLint("JavascriptInterface")
    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_web_view)
        webView = findViewById<WebView>(R.id.webview)
        bundle = getIntent().getExtras()
        checkingStatus = CheckingStatus()
        conn = ConnectivityReceiver()
        val intentFilter = IntentFilter()
        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION)
        registerReceiver(conn, intentFilter)
        checkfunction(this@WebViewActivity)
        getSupportActionBar()!!.setDisplayHomeAsUpEnabled(true)
        if (bundle != null) {
            URL = bundle!!.getString("url")
            webView!!.loadUrl(URL!!)
            webView!!.settings.javaScriptEnabled = true
            webView!!.settings.loadWithOverviewMode = true
            webView!!.settings.useWideViewPort = true
            webView!!.settings.pluginState = WebSettings.PluginState.ON
            webView!!.settings.setRenderPriority(WebSettings.RenderPriority.HIGH)
            webView!!.settings.cacheMode = WebSettings.LOAD_DEFAULT
            webView!!.settings.savePassword = false
            webView!!.settings.domStorageEnabled = true
            webView!!.isHorizontalScrollBarEnabled = false
            webView!!.addJavascriptInterface(this, "Android")
            webView!!.webChromeClient = WebChromeClient()
            webView!!.webViewClient = WebViewClient()
        } else {
            Toast.makeText(this@WebViewActivity, "Please Try Again", Toast.LENGTH_SHORT).show()
            finish()
        }
    }

    fun checkfunction(context: Context?) {
        val isConnected: Boolean = ConnectivityReceiver.Companion.isConnected
        //notification(isConnected,lp_taxi_upto18pass.this);
        checkingStatus!!.notification(isConnected, context!!)
    }

    override fun onNetworkConnectionChanged(isConnected: Boolean) {
        checkingStatus!!.notification(isConnected, this)
    }

    protected override fun onResume() {
        super.onResume()
        MyApplication.Companion.instance!!.setConnectivityListener(this)
    }

    protected override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(conn)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> finish()
        }
        return true
    }
}